### PR TITLE
chore(deps): ensure a dependency workflow checks all upstream resources and dependencies

### PR DESCRIPTION
**Pull request by CollectiveUnicorn**
## Description

- Removes dependabot and replaces it with renovate. Renovate will update all uds-bundles and zarf packages in addition to all the other dependencies that we have. Renovate can be managed here: https://developer.mend.io/ - this site shows all the pending dependency updates, provides the ability to re-run scans, shows what dependencies are detected, and notifies the user if there are any issues with dependency updates. These changes mirror the configs in https://github.com/defenseunicorns/uds-common/blob/main/config/renovate.json5
- You can see this running at: https://github.com/CollectiveUnicorn/leapfrogai-renovate-testing

### BREAKING CHANGES

- Dependabot will no longer function

### CHANGES

- Dependabot has been replaced with renovate to facilitate zarf and uds dependency updates as per the recommendation in #696 

## Related Issue
- Closes: https://github.com/defenseunicorns/leapfrogai/issues/696

## Checklist before merging

- [x] Tests, documentation, ADR added or updated as needed
- [x] Followed the [Contributor Guide Steps](https://github.com/defenseunicorns/leapfrogai/blob/main/.github/CONTRIBUTING.md)

### CODE REVIEW

**Review by justinthelaw — APPROVED** [review]
LGTM! Requires @YrrepNoj review and approval, then activation in the Mend dashboard.

File: renovate.json5
Comment (YrrepNoj) [inline renovate.json5:1]: Are you opposed to putting this in the `.github/` directory?

Comment (CollectiveUnicorn) [reply renovate.json5:1]: No, I'll move it there

Comment (CollectiveUnicorn) [reply renovate.json5:1]: Should be good now

**Review by justinthelaw — APPROVED** [review]

**Review by YrrepNoj — APPROVED** [review]

**Review by jalling97 — APPROVED** [review]

**Pull request merged**